src/CustomHierarchy.cpp

```diff
@@ -626,4 +596,95 @@ template auto CustomHierarchy::asContainerOf<RecordComponent>()
 template auto CustomHierarchy::asContainerOf<Mesh>() -> Container<Mesh> &;
 template auto CustomHierarchy::asContainerOf<ParticleSpecies>()
     -> Container<ParticleSpecies> &;
+
+template <typename KeyType>
+auto CustomHierarchy::bracketOperatorImpl(KeyType &&provided_key)
+    -> mapped_type &
+{
+    auto &cont = container();
+    auto find_special_key =
+        [&cont, &provided_key, this](
+            char const *special_key,
+            auto &alias,
+            auto &&embeddedAccessor) -> std::optional<mapped_type *> {
+        if (provided_key == special_key)
+        {
+            if (auto it = cont.find(provided_key); it != cont.end())
+            {
+                if (it->second.m_attri->get() != alias.m_attri->get() ||
+                    embeddedAccessor(it->second)->m_containerData.get() !=
+                        alias.m_containerData.get())
+                {
+                    /*
+                     * This might happen if a user first creates a custom group
+                     * "fields" and sets the default meshes path as "fields"
+                     * only later.
+                     * If the CustomHierarchy::meshes alias carries no data yet,
+                     * we can just redirect it to that group now.
+                     * Otherwise, we need to fail.
```

Comment:
Scrap this, we should always fail. We can redirect the internal `meshes` handle, but not ones that the user might have as a variable. The joys of dealing with a public member field..